src/module/actor/character/crafting/entry.ts

```diff
@@ -231,6 +232,16 @@ class CraftingEntry implements Omit<CraftingEntryData, "parentItem"> {
     }
 }
 
+interface CraftingEntryRuleData extends RuleElementData {
+    isAlchemical?: boolean;
+    isDailyPrep?: boolean;
+    isPrepared?: boolean;
+    maxItemLevel?: number;
+    maxSlots?: number;
+    craftableItems?: RawPredicate;
+    preparedFormulas?: PreparedFormulaData[];
+}
```

Comment:
Shouldn't it be possible to generate this from the schema?

Reply:
You'd want `SourceFromSchema`

Reply:
oh, didn't see that. but it has the same error 🙁
![image](https://github.com/foundryvtt/pf2e/assets/41452412/ce04e375-f559-4f70-9013-d847ae9cd3c6)

Reply:
make it `Partial`, I guess

Reply:
yeah that works, but it would make `itemUUID` not required anymore which would technically be wrong.

Reply:
partial with an intersection on the picked property

Reply:
should be good now